### PR TITLE
Issue #13: Add support for Azure Open AI

### PR DESCRIPTION
**Pull request by talrejanikhil**
Added Support for Azure Open AI

### CODE REVIEW

File: swarmzero/agent.py [hunk 1]
```diff
@@ -443,6 +444,8 @@ def _assign_agent(self, tools, tool_retriever):
                 agent_class = OpenAIMultiModalLLM
             elif "gpt" in model:
                 agent_class = OpenAILLM
+            elif "azure/" in model:
```
Comment (ToJen) [inline swarmzero/agent.py:447]: great first PR @talrejanikhil!

why the trailing `/` after `azure`?

Comment (talrejanikhil) [reply swarmzero/agent.py:447]: This is to distinguish between regular OpenAI from Azure. In Azure the deployment names could be same as model names and then we will not be able to tell them apart here in this line. This is also a standard way to setup Azure OpenAI in other frameworks :)

**Review by ToJen — APPROVED** [review]

**Comment by talrejanikhil** [comment]
@ToJen I forgot to add the updated poetry lock. Have pushed it now

**Comment by ToJen** [comment]
Thanks for your contribution @talrejanikhil!

**Pull request merged**